tests/framework/prototype.py

```diff
@@ -192,6 +196,25 @@ def assert_rule(self, custom_resource=None, cr_namespace='default'):
             " python -c \"import ray; ray.init(); print(ray.cluster_resources())\"")
         assert rtn == 0
 
+class CurlServiceRule(Rule):
+    """"Using curl to access the deployed application on Ray service"""
+    def assert_rule(self, custom_resource=None, cr_namespace='default'):
+        # Create a pod for running curl command, because the service is not exposed.
+        os.system("kubectl run curl --image=radial/busyboxplus:curl --command --\
+           /bin/sh -c \"while true; do sleep 10;done\"")
+        while True:
+            resp = client.CoreV1Api().read_namespaced_pod(name="curl",
+                                                    namespace='default')
+            if resp.status.phase != 'Pending':
+                break
+            time.sleep(1)
+        output = subprocess.check_output("kubectl exec curl -- \
+            curl -X POST -H 'Content-Type: application/json' \
+            rayservice-sample-serve-svc.default.svc.cluster.local:8000 \
+            -d '[\"MANGO\", 2]'", shell=True)
```

Comment:
What's the meaning of `shell=True`?

Reply:
Setting the shell argument to a true value causes subprocess to spawn an intermediate shell process, and tell it to run the command. Besides, the command should be split a list of arguments instead of a string if we do not use ```shell=True```. The two lines below is simple commands to show how to call the function with or without setting show to True.
```subprocess.check_output("ls -l", shell=True)```
```subprocess.check_output(["ls",  "-l"])```